### PR TITLE
LLVM 11.0.0

### PR DESCRIPTION
**Pull request by liushuyu**
Topic Description
-----------------

Update LLVM to 11.

Package(s) Affected
-------------------

```
extra-devel/ccls
extra-devel/spirv-llvm-translator
extra-libs/opencl-clang
extra-devel/intel-graphics-compiler
extra-devel/pocl
extra-devel/ispc
extra-devel/cquery
extra-dlang/ldc
extra-libs/libobjc2
extra-libs/mesa
extra-libs/qt-5
extra-libs/pyside2
extra-rust/rustc
extra-devel/codelite
extra-devel/intellij-idea
extra-gnome/gnome-builder
extra-devel/qtcreator
extra-kde/kdevelop
extra-enlightenment/edi
extra-admin/atm
extra-admin/ytop
extra-devel/sccache
extra-devel/tokei
extra-doc/tealdeer
extra-editors/neovim-gtk
extra-gnome/fractal
extra-i18n/ydcv-rs
extra-java/icedtea-web
extra-libs/js-68
extra-libs/js-78
extra-libs/librsvg
extra-multimedia/netease-cloud-music-gtk
extra-network/shadowsocks-rust
extra-phosh/squeekboard
extra-python/rustpython
extra-rust/cargo-audit
extra-rust/racer
extra-utils/alacritty
extra-utils/autojump-rs
extra-utils/bat
extra-utils/exa
extra-utils/fd
extra-utils/mcfly
extra-utils/nix-user-chroot
extra-utils/ripgrep
extra-web/zola
```

Security Update?
----------------

No

Build Order
-----------

See above

Architectural Progress
----------------------

- [X] AMD64 `amd64`   
- [X] AArch64 `arm64`
- [X] 32-bit Optional Environment `optenv32`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [X] Loongson 3 `loongson3`
- [X] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
- [ ] 32-bit Optional Environment `optenv32`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

---
This change is [Reviewable](https://reviewable.io/reviews/aosc-dev/aosc-os-abbs/2635)

### CODE REVIEW

**Comment by liushuyu** [comment]
Supersedes #2547.

File: extra-devel/intel-graphics-compiler/autobuild/prepare [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #rm -v "$SRCDIR"/IGC/VectorCompiler/spirv-patches-new/*.patch
 
 abinfo "Patching SPRIV-LLVM-Translator using AOSC-patched Intel patches ..."
-PATCHES_ROOT="$(readlink -f "$SRCDIR"/IGC/VectorCompiler/spirv-patches-10/)"
+PATCHES_ROOT="$(readlink -f "$SRCDIR"/IGC/VectorCompiler/spirv-patches-11/)"
```
Comment (MingcongBai) [inline extra-devel/intel-graphics-compiler/autobuild/prepare:6]: Could probably use `llvm-config` here for version generation?

File: extra-devel/intel-graphics-compiler/spec [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
-      git::commit=c8c52b5fb14b33e32de9df573b7de186a0c97c94;rename=vc-intrinsics::https://github.com/intel/vc-intrinsics \
-      git::commit=e8a52ab21a3c2244b56b444c29ba0b65c7a7f4fd;rename=SPIRV-LLVM-Translator::https://github.com/KhronosGroup/SPIRV-LLVM-Translator"
-CHKSUMS="sha256::0cf353208648c15ede817095de5fa245efc2617d823b918e575208bccfa36ebe \
+      git::commit=50326439b1d0733c6af697de856d922273e4cfbe;rename=vc-intrinsics::https://github.com/intel/vc-intrinsics \
+      git::commit=d6dc999eee381158a26f048a333467c9ce7e77f2;rename=SPIRV-LLVM-Translator::https://github.com/KhronosGroup/SPIRV-LLVM-Translator"
+CHKSUMS="sha256::9bf29ef4c70ff7951f67861c21825e5f341716b8cc48af29aef3fc023845d7a7 \
          SKIP \
          SKIP"
 SUBDIR="intel-graphics-compiler-igc-$VER"
+REL=0
```
Comment (MingcongBai) [inline extra-devel/intel-graphics-compiler/spec]: Drop this line.

File: extra-devel/ispc/spec [hunk 3]
```diff
@@ -1,4 +1,4 @@
 VER=1.14.1
-REL=2
+REL=4
```
Comment (MingcongBai) [inline extra-devel/ispc/spec:2]: Wrong bump?

File: extra-devel/qtcreator/spec [hunk 4]
```diff
@@ -1,3 +1,4 @@
-VER=4.13.2
+VER=4.13.3
 SRCS="tbl::https://download.qt.io/official_releases/qtcreator/${VER:0:4}/$VER/qt-creator-opensource-src-$VER.tar.xz"
-CHKSUMS="sha256::b00a5c13ba67d2e3436e6d9bf32fba38d780a1de0bc85b813daac672021e9428"
+CHKSUMS="sha256::7b0317ca9354284e98967e94a30a1b813db379017d28d9847847498c113df9e5"
+REL=0
```
Comment (MingcongBai) [inline extra-devel/qtcreator/spec]: Drop this line.

File: extra-dlang/dub/spec [hunk 5]
```diff
@@ -1,4 +1,4 @@
-VER=1.18.0
+VER=1.23.0
 SRCS="https://github.com/dlang/dub/archive/v$VER.tar.gz"
-CHKSUMS="sha256::5ea118388217ad9afe7ccd6a486c0139c39a45e464de662fecfa142a408c1880"
-REL=1
+CHKSUMS="sha256::9c7196fd76efce50e8606b3f0ab103bb8b3984842dcc7c0970eb07aba63dbaa5"
+REL=0
```
Comment (MingcongBai) [inline extra-dlang/dub/spec]: Drop this line.

File: extra-dlang/ldc/spec [hunk 6]
```diff
@@ -1,4 +1,4 @@
-VER=1.23.0
-REL=3
-SRCS="https://github.com/ldc-developers/ldc/releases/download/v${VER/b/-beta}/ldc-${VER/b/-beta}-src.tar.gz"
-CHKSUMS="sha256::6d18d233fb3a666113827bdb7d96a6ff0b54014bbeb76d0cd12a892e8490afb9"
+VER=1.24.0
+SRCS="tbl::https://github.com/ldc-developers/ldc/releases/download/v${VER/b/-beta}/ldc-${VER/b/-beta}-src.tar.gz"
+CHKSUMS="sha256::fd9561ade916e9279bdcc166cf0e4836449c24e695ab4470297882588adbba3c"
+REL=0
```
Comment (MingcongBai) [inline extra-dlang/ldc/spec]: Drop this line.

File: extra-gnome/fractal/spec [hunk 7]
```diff
@@ -1,4 +1,4 @@
-VER=4.0.0
-REL=2
-SRCTBL="https://github.com/danigm/fractal/archive/$VER.tar.gz"
-CHKSUM="sha256::f55b01dba5512abedd907d65bf1cd3a96243d81004152cd681ce7e306f777c2f"
+VER=4.4.0
+REL=0
```
Comment (MingcongBai) [inline extra-gnome/fractal/spec]: Drop this line.

File: extra-kde/kdevelop-pg-qt/spec [hunk 8]
```diff
@@ -1,4 +1,4 @@
-VER=2.1.0
-REL=2
-SRCTBL="https://download.kde.org/stable/kdevelop-pg-qt/$VER/src/kdevelop-pg-qt-$VER.tar.xz"
-CHKSUM="sha256::20d84d25bd40610bd6c0964e4fe0642e56c41b76a65575122dc5196649621e5d"
+VER=2.2.1
+REL=0
```
Comment (MingcongBai) [inline extra-kde/kdevelop-pg-qt/spec]: Drop this line.

File: extra-kde/kdevelop/spec [hunk 9]
```diff
@@ -1,4 +1,4 @@
-VER=5.5.2
-REL=3
-SRCTBL="https://download.kde.org/stable/kdevelop/$VER/src/kdevelop-$VER.tar.xz"
-CHKSUM="sha256::228e981a268704fef0f8fa437e37e3e2e0d261d3a2afbabff0f4d060c31f74da"
+VER=5.6.1
+REL=0
```
Comment (MingcongBai) [inline extra-kde/kdevelop/spec]: Drop this line.

File: extra-libs/libobjc2/spec [hunk 10]
```diff
@@ -1,4 +1,4 @@
-VER=1.8.1
-SRCTBL="https://github.com/gnustep/libobjc2/archive/v$VER.tar.gz"
-CHKSUM="sha256::472bd684070c4dc6e78293836f426c1c6981c3818b546f74ec400bb8f13df925"
-REL=4
+VER=2.1
+SRCS="git::commit=tags/v$VER::https://github.com/gnustep/libobjc2"
+CHKSUMS="SKIP"
+REL=0
```
Comment (MingcongBai) [inline extra-libs/libobjc2/spec]: Drop this line.

File: extra-lua/lua-mpack/autobuild/build [hunk 11]
```diff
@@ -1,3 +1,5 @@
+set -x
 gcc ${CPPFLAGS} ${CFLAGS} ${LDFLAGS} -DMPACK_USE_SYSTEM -I/usr/include/lua5.1 -c lmpack.c -o lmpack.o
-gcc ${CPPFLAGS} ${CFLAGS} ${LDFLAGS} -lmpack -shared -o mpack.so lmpack.o
+gcc ${CPPFLAGS} ${CFLAGS} ${LDFLAGS} -o mpack.so lmpack.o -lmpack -shared
```
Comment (MingcongBai) [inline extra-lua/lua-mpack/autobuild/build]: Use absolute paths here.

File: extra-network/shadowsocks-rust/spec [hunk 12]
```diff
@@ -1,4 +1,4 @@
-VER=1.8.16
-REL=1
+VER=1.8.23
+REL=0
```
Comment (MingcongBai) [inline extra-network/shadowsocks-rust/spec]: Drop this line.

File: extra-optenv32/glibc+32/autobuild/build [hunk 13]
```diff
@@ -28,10 +28,6 @@ BUILD_CC="gcc" CC="i686-pc-linux-gnu-gcc" \
 make
 make install_root="$PKGDIR" install 
 
-cp -v ../sunrpc/rpc/*.h "$PKGDIR"/opt/32/include/rpc
-cp -v ../sunrpc/rpcsvc/*.h "$PKGDIR"/opt/32/include/rpcsvc
-cp -v ../nis/rpcsvc/*.h "$PKGDIR"/opt/32/include/rpcsvc
-
 cd ..
```
Comment (MingcongBai) [inline extra-optenv32/glibc+32/autobuild/build:35]: Use absolute path here (throughout the script).

File: extra-optenv32/llvm+32/autobuild/build [hunk 14]
```diff
@@ -16,8 +16,8 @@ cmake .. -DCMAKE_INSTALL_PREFIX=/opt/32 -DCMAKE_BUILD_TYPE=Release \
          -DLLVM_BINUTILS_INCDIR=/opt/32/include \
          -DCOMPILER_RT_DEFAULT_TARGET_TRIPLE=i686-pc-linux-gnu -DLLVM_DEFAULT_TARGET_TRIPLE=i686-pc-linux-gnu -DLLVM_HOST_TRIPLE=i686-pc-linux-gnu
 
-make
-make install DESTDIR="$PKGDIR"
+ninja
+DESTDIR="$PKGDIR" ninja install
 
 cd $SRCDIR
```
Comment (MingcongBai) [inline extra-optenv32/llvm+32/autobuild/build:22]: Quote variables.

File: extra-rust/cargo-audit/spec [hunk 15]
```diff
@@ -1,4 +1,4 @@
-VER=0.12.0
-REL=1
+VER=0.13.1
+REL=0
```
Comment (MingcongBai) [inline extra-rust/cargo-audit/spec]: Drop this line.

File: extra-rust/racer/spec [hunk 16]
```diff
@@ -1,4 +1,4 @@
-VER=2.1.38
-REL=1
-SRCTBL="https://github.com/phildawes/racer/archive/v$VER.tar.gz"
-CHKSUM="sha256::e118e35915fee2b8d77d8de3c1624504778991e8d8bf8efd2934acaa88070f3f"
+VER=2.1.40
+REL=0
```
Comment (MingcongBai) [inline extra-rust/racer/spec]: Drop this line.

File: extra-utils/mcfly/spec [hunk 17]
```diff
@@ -1,4 +1,4 @@
-VER=0.3.3
+VER=0.5.0
 GITCO="tags/v$VER"
 GITSRC="https://github.com/cantino/mcfly/"
-REL=1
+REL=0
```
Comment (MingcongBai) [inline extra-utils/mcfly/spec]: Drop this line.

File: extra-utils/tilix/spec [hunk 18]
```diff
@@ -1,3 +1,4 @@
-VER=20190901
-REL=3
-SRCS="git::commit=e302c58e7ec262496dce27db3e4adc394caf457d::https://github.com/gnunn1/tilix"
+VER=20200820
+REL=0
```
Comment (MingcongBai) [inline extra-utils/tilix/spec]: Drop this line.

File: groups/llvm-rebuilds [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+extra-libs/mesa
+extra-libs/qt-5
+extra-libs/pyside2
+extra-rust/rustc
+extra-devel/codelite
+extra-devel/intellij-idea
+extra-gnome/gnome-builder
+extra-devel/qtcreator
+extra-kde/kdevelop
+extra-enlightenment/edi
```
Comment (MingcongBai) [inline groups/llvm-rebuilds:19]: Could do with an update against the build order you specified in the PR?

File: extra-devel/ccls/spec [hunk 20]
```diff
@@ -1,4 +1,4 @@
-VER=0.20190823.6
+VER=0.20201025
 GITSRC="https://github.com/MaskRay/ccls"
 GITCO="tags/$VER"
-REL=1
+REL=0
```
Comment (MingcongBai) [inline extra-devel/ccls/spec]: Drop this line.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-devel/intel-graphics-compiler/autobuild/prepare (hunk 1 above)
Comment (liushuyu) [reply extra-devel/intel-graphics-compiler/autobuild/prepare:6]: No, Intel does not work this way, the folder name could be `spirv-patches-new` or something.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**